connexion/validators/abstract.py

```diff
@@ -99,11 +99,14 @@ def _insert_messages(
         receive: Receive, *, messages: t.Iterable[t.MutableMapping[str, t.Any]]
     ) -> Receive:
         """Insert messages at the start of the `receive` channel."""
+        # Ensure that messages in an iterator
+        messages = iter(messages)
 
         async def receive_() -> t.MutableMapping[str, t.Any]:
-            for message in messages:
-                return message
-            return await receive()
+            try:
+                return next(iter(messages))
```

Comment:
There's double `iter()` call (also done on line 103), here we can just do `next(messages)` I believe